logger.tf

```diff
@@ -0,0 +1,50 @@
+locals {
+  api_mgmt_name = var.apim_suffix == "" ? "cft-api-mgmt-${var.environment}" : "cft-api-mgmt-${var.apim_suffix}"
+  api_mgmt_resource_group = "cft-${var.environment}-network-rg"
+  api_mgmt_logger_name = "${local.api_mgmt_name}-logger"
```

Comment:
api_mgmt_logger_name can be replaced with ${local.name}-logger.  You'll need to do this if you remove the api_mgmt_name local variable as suggested in another comment.